cogs/mod.py

```diff
@@ -111,6 +165,48 @@ async def finalize_crabbo(self, ctx):
 
         await ctx.message.delete()
 
+    async def verification_purge_utility(self, after):
+        # after can be None, a datetime or a discord message
+        await self.verification_channel.send("Starting verification purge")
+        channel = self.verification_channel
+        deleted = 0
+        async for message in channel.history(oldest_first=True, limit=None, after=after):
+            if len(message.attachments) > 0 or len(message.embeds) > 0:
+                content = message.content
+                if len(message.embeds) > 0:
+                    thumbnail_found = False
+                    for embed in message.embeds:
+                        if embed.thumbnail:
+                            thumbnail_found = True
+                            content = content.replace(embed.thumbnail.url, "")
+                    if not thumbnail_found:
+                        continue
+                if content:
+                    await channel.send(
+                        f"{message.author} sent the following purged message on "
+                        f"{message.created_at.strftime('%Y/%m/%d, %H:%M:%S')}: {content}"
+                    )
+                await message.delete()
+                deleted += 1
+
+        await self.verification_channel.send(
+            f"Verification purge completed. Deleted {deleted} message{'s' if deleted != 1 else ''}"
+        )
+
+    @commands.command()
+    @is_moderator()
+    async def verification_purge(self, ctx, id: int = None):
+        """ "
+        Manually start the purge of pictures in the verification channel.
+
+        If a message ID is provided, every pictures after that message will be removed.
+        If no message ID is provided, this will be done for the whole channel (may take time).
+        """
+        message = None
+        if id:
```

Comment:
i'd prefer `id is not None` here since id can technically also be 0 although this would never happen here, but as a pattern for ints explicitly checking None is better practice

Reply:
Resolved in 02632a4